src/governance/timelock/timelock_controller.cairo

```diff
@@ -0,0 +1,660 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.14.0 (governance/timelock/timelock_controller.cairo)
+
+/// # TimelockController Component
+///
+/// Component that acts as a timelocked controller. When set as the owner of an `Ownable` smart contract,
+/// it enforces a timelock on all `only_owner` maintenance operations. This gives time for users
+/// of the controlled contract to exit before a potentially dangerous maintenance operation is applied.
+///
+/// By default, this component is self administered, meaning administration tasks have to go through
+/// the timelock process. The proposer role is in charge of proposing operations. A common use case
+/// is to position the timelock controller as the owner of a smart contract, with a multi-sig
+/// or a DAO as the sole proposer.
+#[starknet::component]
+pub mod TimelockControllerComponent {
+    use core::hash::{HashStateTrait, HashStateExTrait};
+    use core::num::traits::Zero;
+    use core::poseidon::PoseidonTrait;
+    use openzeppelin::access::accesscontrol::AccessControlComponent::InternalTrait as AccessControlInternalTrait;
+    use openzeppelin::access::accesscontrol::AccessControlComponent::{
+        AccessControlImpl, AccessControlCamelImpl
+    };
+    use openzeppelin::access::accesscontrol::AccessControlComponent;
+    use openzeppelin::access::accesscontrol::DEFAULT_ADMIN_ROLE;
+    use openzeppelin::governance::timelock::interface::{ITimelock, TimelockABI};
+    use openzeppelin::governance::timelock::utils::OperationState;
+    use openzeppelin::governance::timelock::utils::call_impls::{HashCallImpl, Call};
+    use openzeppelin::introspection::src5::SRC5Component::SRC5Impl;
+    use openzeppelin::introspection::src5::SRC5Component;
+    use starknet::ContractAddress;
+    use starknet::SyscallResultTrait;
+
+    // Constants
+    pub const PROPOSER_ROLE: felt252 = selector!("PROPOSER_ROLE");
+    pub const EXECUTOR_ROLE: felt252 = selector!("EXECUTOR_ROLE");
+    pub const CANCELLER_ROLE: felt252 = selector!("CANCELLER_ROLE");
+    pub const DONE_TIMESTAMP: u64 = 1;
+
+    #[storage]
+    struct Storage {
+        TimelockController_timestamps: LegacyMap<felt252, u64>,
+        TimelockController_min_delay: u64
+    }
+
+    #[event]
+    #[derive(Drop, PartialEq, starknet::Event)]
+    pub enum Event {
+        CallScheduled: CallScheduled,
+        CallExecuted: CallExecuted,
+        CallSalt: CallSalt,
+        Cancelled: Cancelled,
+        MinDelayChange: MinDelayChange
```

Comment:
```suggestion
        MinDelayChanged: MinDelayChanged
```
I know this was for consistency with Solidity, but we should be consistent with our event names, and it shouldn't be a big problem for indexers.

Reply:
Also what about `CallCancelled`? For some reason, CallSalt doesn't bother me though.

Reply:
There's no verb in `CallSalt`; otherwise, we could do `CallSaltAdded`. I agree `CallSalt` is good as is though